### PR TITLE
Delegated harvesting

### LINKED ISSUE #468: [Guide] Add Node and VRF transactions to delegated havesting guide
The guide [Activating delegated harvesting](https://nemtech.github.io/guides/accountlink/activating-delegated-harvesting.html)  and concept [Harvesting](https://docs.symbolplatform.com/concepts/harvesting.html) is outdated.

Due to finality changes, an account must send first a **NodeKeyLinkTransaction** and a **VrfKeyLinkTransaction** in order to activate delegated harvesting. The AccountLinkTransaction and the new transaction types can be announced together in an AggregateTransaction.

To update the guide and code examples, refer to the [Symbol SDK Integration Test](https://github.com/nemtech/symbol-sdk-typescript-javascript/blob/main/e2e/infrastructure/PersistentHarvesting.spec.ts).

### LINKED ISSUE #532: Activating delegated harvesting example needs to be updated.
**Describe the error**

In the example it states that node certificate public can be retrieve using node info.  This is not correct.  Node info returns the CA certificate public for the node.
https://docs.symbolplatform.com/guides/accountlink/activating-delegated-harvesting.html#method-01-using-the-sdk

> Note
> Get the node’s public key by querying http://<node-url>:3000/node/info.

To get the node certificate public, it has to be provide by the node owner.  Below is a list of node certificate public keys for the testnet harvesting servers. 
| Host name                                                         | Node publicKey                                                                       |
|:----------------------------------------------------|:--------------------------------------------------------------------|
| beacon-01.ap-northeast-1.0.10.0.x.symboldev.network | be60be426872b3cb46fe2c9baa521731ea52c0d57e004fc7c84293887ac3bad0    |
| beacon-01.eu-central-1.0.10.0.x.symboldev.network   | ee356a555802003c5666d8485185cdc3844f9502faf24b589bdc4d6e9148022f    |
| beacon-01.us-west-2.0.10.0.x.symboldev.network      | 81890592f960aaebda7612c8917fa9c267a845d78d74d4b3651af093e6775001    |
| beacon-01.ap-southeast-1.0.10.0.x.symboldev.network | 2af52c5aa9a5e13dd548a577debf21e7d3cc285a1b0798f4d450239cdde5a169    |
| beacon-01.us-west-1.0.10.0.x.symboldev.network      | d74b89ee9378debd510a4139f8e8b10b878e12956059cd9e13253cf3ad73bdeb    |
| beacon-01.eu-west-1.0.10.0.x.symboldev.network      | 938d6c1bbdb09f3f1b9d95d2d902a94c95e3aa6f1069a805831d9e272dcf927f    |
| beacon-01.us-east-1.0.10.0.x.symboldev.network      | 2a40f7895f56389be40c063b897e9e66e64705d55b19fc43c8ceb5f7f14abe59    |

### LINKED ISSUE #547: [Guide] Local harvesting through a remote account
I don't think I was aware of this option, but it should be explained somewhere.

https://nem2.slack.com/archives/C9YKR0EUX/p1608225259204300?thread_ts=1608214610.203500&cid=C9YKR0EUX
https://nemgroup-workspace.slack.com/archives/C015AEQ9ZL2/p1608230255400000?thread_ts=1608219011.393000&cid=C015AEQ9ZL2

### PR DESCRIPTION
**Pull request by segfaultxavi**
Updated the Guide and the Concept.

Fixes #468
Fixes #532
Fixes #547

### CODE REVIEW

File: source/guides/accountlink/activating-delegated-harvesting.rst [hunk 1]
```diff
@@ -12,162 +12,128 @@ Activating delegated harvesting
...
-*************************
-Method #01: Using the SDK
-*************************
+- A **Remote account (R)** that will act as a proxy between **M** and the node. This account **must have never sent or received any transaction**, and it cannot be involved in any transaction while it is a delegated account.
 
-1. Define your **main account (M)**, **announcer account(A)** and the **remote account (R)** using their private keys.
+- A **VRF account (V)** that has never sent or received any transactions. It is a regular account used to add randomness to the account selection process.
 
-.. example-code::
+- The **node's public TLS key**. This is the key the node uses to authenticate data for transport over TLS and must be provided by the node owner, it cannot be retrieved remotely through any |codename| API.
```
Comment (rg911) [inline source/guides/accountlink/activating-delegated-harvesting.rst:47]: For Dual Nodes which is using the catapult-rest >2.2.0, the node's public key can be retrieved from the `node/info` endpoint. For peer only nodes, the key can be extracted from tls connection handshake if the ca can be trusted..

File: source/guides/accountlink/activating-delegated-harvesting.rst [hunk 2]
```diff
@@ -12,162 +12,128 @@ Activating delegated harvesting
...
-    :caption: Delegated harvesting activation diagram
-    :alt: Delegated harvesting activation diagram
-    :align: center
+2. Link the main account **M** to a **VRF account (V)** for randomized block production and account selection using a :ref:`VrfKeyLinkTransaction <vrf-key-link-transaction>`.
+
+3. Link the main account **M** to a node in order to harvest through that node using a :ref:`NodeKeyLinkTransaction <node-key-link-transaction>`.
+
+4. Request the node to add the remote account **R** as a delegated harvester using a :ref:`PersistentDelegationRequestTransaction <persistent-delegation-request-transaction>`.
+
+Please note that it is entirely up to the node to comply with the request. Moreover, **by design** there is no mechanism to check if the request has been accepted or not, since there is no way to provide such mechanism in a trustless environment.
```
[the remark this answers is not in the document]
Comment (segfaultxavi) [reply source/guides/accountlink/activating-delegated-harvesting.rst]: I've added this information to the guide.

File: source/guides/accountlink/activating-delegated-harvesting.rst (hunk 1 above)
Comment (segfaultxavi) [reply source/guides/accountlink/activating-delegated-harvesting.rst]: I've added this information to the guide.

**Comment by segfaultxavi** [comment]
This has been under review for way too long without activity and feedback on Slack was positive so... merging.

**Pull request merged**